python/common/org/python/types/Int.java

```diff
@@ -78,8 +78,22 @@ public Int(org.python.Object[] args, java.util.Map<java.lang.String, org.python.
                     );
                 }
             }
-        } else if (args.length > 3) {
-            throw new org.python.exceptions.NotImplementedError("int() with a base is not implemented");
+        } else if (args.length == 2) {
+            try {
+                org.python.types.Str argStr = (org.python.types.Str) args[0];
+                org.python.types.Int argInt = (org.python.types.Int) args[1];
+                int base = (int) argInt.value;
+                if (base == 0) {
+                    base = 10;
+                } else if (base < 2 || base > 36) {
+                    throw new org.python.exceptions.ValueError("int() base must be >= 2 and <= 36, or 0");
```

Comment:
Almost there, now the tests are failing because the message isn't supposed to contain `, or 0` at the end.
The `0` here is considered as `falsy` value, e.g. `bool(0)` (you might add a test for the `None` case too).

Reply:
Running it under Python 3.4.8 has the `, or 0` at the end. I'll add a switch for the version.

Reply:
Are you sure?

```
$ $(pythonz locate 3.4.8)
Python 3.4.8 (default, Jun  9 2018, 11:17:16) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> int('22', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: int() base must be >= 2 and <= 36
```

```
$ $(pythonz locate 3.5.0)
Python 3.5.0 (default, Nov  4 2015, 10:43:10) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> int('22', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: int() base must be >= 2 and <= 36
>>> 
```

```
$ $(pythonz locate 3.6.0)
Python 3.6.0 (default, Feb  6 2017, 21:07:35) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> int('22', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: int() base must be >= 2 and <= 36
```

Reply:
Then it must be one of the minor versions under 3.6 that changed that behavior. I'll check which one it is.

Reply:
The error message changed from 3.6.3 to 3.6.4